CMakeLists.txt

```diff
@@ -14,6 +14,37 @@ set(INSTALL_CMAKE_DIR lib/cmake CACHE PATH "Installation directory for cmake fil
 
 set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
 
+set(CTEST_MEMORYCHECK_TYPE "Valgrind")
+
+set(MEMORYCHECK_COMMAND_OPTIONS_REGULAR " \
```

Comment:
This variable name is a bit long, maybe `VALGRIND_DEFAULT` and `VALGRIND_PEDANTIC` instead of `MEMORYCHECK`, because these are valgrind specific options.

Reply:
Also, could we define `VALGRIND_PEDANTIC` using `VALGRIND_DEFAULT`?
I'm imagining something like
```
VALGRIND_PEDANTIC = VALGRIND_DEFAULT + "--errors-for-leak-kinds=all"
```

I think that would make it a lot easier to see what they are doing differently.

Reply:
I have addressed this comment. I agree that we can make the variable name shorter and concise.

The exact syntax to set up `VALGRIND_PEDANTIC` to `VALGRIND_DEFAULT` is
`set(VALGRIND_PENDANTIC "${VALGRIND_DEFAULT} --errors-for-leak-kinds=all")`.
https://stackoverflow.com/questions/18001198/how-do-you-concatenate-string-in-cmake